### ISSUE TITLE
CA1822 false positve with WPF UserControl

### ISSUE THREAD

**Issue by bdovaz** [dotnet/roslyn-analyzers#6692]
### Analyzer

**Diagnostic ID**: CA1822

### Analyzer source

**Version**: 7.0.304

### Describe the bug

Executing this command with a class that has the following code creates a false positive.

`dotnet format --severity info MySolution.sln`

Before:

```csharp
public partial class TestUserControl : UserControl
{
    public TestUserControl()
    {
        InitializeComponent();
    }

    private void MyEventHandler()
    {
        MyControl.Visibility = Visibility.Hidden;
    }
}
```

After:

```csharp
public partial class TestUserControl : UserControl
{
    public TestUserControl()
    {
        InitializeComponent();
    }

    private static void MyEventHandler()
    {
        MyControl.Visibility = Visibility.Hidden;
    }
}
```
